src/thermo/CoverageDependentSurfPhase.cpp

```diff
@@ -0,0 +1,571 @@
+                        item.second->input, "Unrecognized coverage dependency model. \
+                        Model must be 'linear', 'piecewise-linear', 'polynomial', \
+                        or 'interpolative'.");
+                }
+                // For coverage-dependent heat capacity parameters, if present
+                if (cov_map2.hasKey("heat-capacity-a")) {
+                    HeatCapacityDependency cpcov_deps(k, j);
+                    cpcov_deps.coeff_a = cov_map2.convert("heat-capacity-a", "J/kmol/K");
+                    cpcov_deps.coeff_b = cov_map2.convert("heat-capacity-b", "J/kmol/K");
+
+                    cp_ind = m_HeatCapacityDependency.size();
+                    m_indexmap_cp[item.first][item2.first] = cp_ind;
+                    m_HeatCapacityDependency.push_back(cpcov_deps);
+                }
+            }
+        }
+    }
+}
+
+bool CoverageDependentSurfPhase::addSpecies(shared_ptr<Species> spec)
+{
+    bool added = SurfPhase::addSpecies(spec);
+    if (added) {
+        m_cov.push_back(0.0);
+        m_h_cov.push_back(0.0);
+        m_s_cov.push_back(0.0);
+        m_cp_cov.push_back(0.0);
+        m_mu_cov.push_back(0.0);
+        m_enthalpy.push_back(0.0);
+        m_entropy.push_back(0.0);
+        m_heatcapacity.push_back(0.0);
+        m_chempot.push_back(0.0);
+    }
+    return added;
+}
+
+void CoverageDependentSurfPhase::getParameters(AnyMap& phaseNode) const
+{
+    SurfPhase::getParameters(phaseNode);
+    phaseNode["reference-state-coverage"] = m_theta_ref;
+}
+
+void CoverageDependentSurfPhase::getSpeciesParameters(const std::string& name,
+                                                      AnyMap& speciesNode) const
+{
+    SurfPhase::getSpeciesParameters(name, speciesNode);
+    // Get linear model parameters from PolynomialDependency vector
+    if (m_indexmap_lin.count(name)) {
+        for (const auto& item: m_indexmap_lin.at(name)) {
+            auto& covdepNode =
+                speciesNode["coverage-dependencies"][item.first].getMapWhere(
+                    "model", "linear", true);
+            covdepNode["enthalpy"].setQuantity(
+                m_PolynomialDependency[item.second].enthalpy_coeffs[1], "J/kmol");
+            covdepNode["entropy"].setQuantity(
+                m_PolynomialDependency[item.second].entropy_coeffs[1], "J/kmol/K");
+        }
+    }
+    // Get polynomial model parameters from PolynomialDependency vector
+    if (m_indexmap_poly.count(name)) {
+        for (const auto& item: m_indexmap_poly.at(name)) {
+            auto& covdepNode =
+                speciesNode["coverage-dependencies"][item.first].getMapWhere(
+                    "model", "polynomial", true);
+            vector_fp hvec =
+                vector_fp(
+                    m_PolynomialDependency[item.second].enthalpy_coeffs.begin() + 1,
+                    m_PolynomialDependency[item.second].enthalpy_coeffs.end());
```

Comment:
This can be written a little bit more simply as just
```suggestion
            vector_fp hvec(
                m_PolynomialDependency[item.second].enthalpy_coeffs.begin() + 1,
                m_PolynomialDependency[item.second].enthalpy_coeffs.end());
```